### DISCUSSION TITLE
argocd login --core and RBAC

### DISCUSSION THREAD

**Discussion by jritter** [argoproj/argo-cd#8280] [Q&A] [answered]
Hi folks,

I just poked around with the argocd cli tool, and stumbled across the --core login option, which allowed me to access all the cli tool goodness through the Kubernetes API. 

What I couldn't wrap my head around though is how ArgoCD RBAC [1] is honored in this scenario. In our setup (ArgoCD configured using Dex OAuth on OpenShift 4.8), we allow the members of a certain group to read all the applications (role:readonly), and members of another group to perform write tasks such as syncing applications (role:admin). This works in the Web UI as expected, after having logged in using Dex OAuth.

After doing argocd login --core while logged in using the read-only user, I was able to perform an application sync (argocd app sync <appname>). This should not work according to the ArgoCD RBAC policy. I have to mention that this user has admin privileges on the namespace where argocd is deployed, and I suppose that plays a role here.

Can anybody explain how this is supposed to work? I wasn't able to find any documentation on that.

-- Juerg

[1] https://argo-cd.readthedocs.io/en/stable/operator-manual/rbac/

**Comment by mikebryant** [comment, marked as answer]
> What I couldn't wrap my head around though is how ArgoCD RBAC [1] is honored in this scenario.

It won't be. The CLI using the Kubernetes API directly will mean only the Kubernetes RBAC is in effect.

> I have to mention that this user has admin privileges on the namespace where argocd is deployed, and I suppose that plays a role here.

This is exactly right. If the user has full access to all object types in the Namespace, they can do anything. Which CLI tool they're using for this (kubectl or argocd cli) makes no difference to their access to the API.

> Can anybody explain how this is supposed to work? I wasn't able to find any documentation on that.

I think this is mostly a convenience tool for ArgoCD admins - if they already have Kubernetes API access they can use that using the argocd domain-specific tooling instead of `kubectl edit`

**Reply by jritter** [reply]
Thanks MIke, that helps. as long as we can prevent this ArgoCD RBAC circumvention by locking down the argocd namespace we don't see a problem with it.

**Comment by sabinayakc** [comment]
@jritter  how did you get the `login --core` to work. I can't seem to login. 

https://github.com/argoproj/argo-cd/discussions/12302
